File: backend/DS.Api/DS.Core/Models/FilterModel/FilterModel.cs
namespace DS.Core.Models.FilterModel
{
    public class FilterModel
    {
        public string? FilterKey { get; set; }
        public string Sort { get; set; } = string.Empty;
        public string Order { get; set; } = string.Empty;
        public int PageSize { get; set; } = 10;
        public int PageIndex { get; set; }

        public string SortExpression()
        {
            return $"{Sort} {Order}";
        }

        public int RecordToSkip()
        {
            if (PageSize == 0)
            {
                PageSize = 10;
            }
            return PageIndex * PageSize;
        }
    }
}

File: frontend/src/app/shared/models/filter.model.ts
export class FilterModel{
    filterKey:string;
    pageSize:number;
    pageIndex:number;
    sort:string;
    order:string;

    constructor(){
        this.order = "asc";
        this.pageIndex = 0;
        this.pageSize = 10;
    }

    toQueryString(): string {
        let queryString = `sort=${this.sort}&order=${this.order}&pageIndex=${this.pageIndex}&pageSize=${this.pageSize}`;

        if (this.filterKey !== '' && this.filterKey !== null && this.filterKey !== undefined) {
            queryString += '&filterKey=' + this.filterKey;
        }
        return queryString;
    }
}